### PR TITLE
Add API documentation generation and use Markdown

### PR DESCRIPTION
**Pull request by b0661**
Add generation of the API documentation for akkudoktoreos and akkudoktoreosserver packages.
- The API documentation is generated by the Sphinx autosummary and autodoc extensions. The autodoc extension extracts the API info from the [Sphinx type docstrings](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html) or [Google type docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) of the Python packages.
- The API documentation is also checked by ruff rules. Docstrings are currently not enforced to keep the current code base without change. Minor documentation quirks were adapted to the ruff rules. 

Add documentation generation for the server interface
- Currently this is a dummy description taken from #163 as EOS will switch to fastAPI. 

Move most of the documentation to Markdown format
- Markdown was regarded easier to edit and handle (see Issue #181).

Documentation generation emits some warnings that I could not solve. This does not prevent the creation of the API doc.

You may have a look in the documentation by:

```bash
make read-docs
```

### CODE REVIEW

**Comment by drbacke** [comment]
@b0661 Please resolve the conflicts. Then I will merge it directly. Very helpfull

**Comment by b0661** [comment]
@drbacke Conflicts reolved

**Comment by drbacke** [comment]
Please re-run all tests, code changed quite a lot

**Comment by b0661** [comment]
> Please re-run all tests, code changed quite a lot

@drbacke, tests passed.

File: src/akkudoktoreos/class_ems.py [hunk 1]
```diff
@@ -54,12 +54,11 @@ def simuliere_ab_jetzt(self) -> dict:
...
-        """
-        hour:
-            akku_soc_pro_stunde begin of the hour, initial hour state!
-            last_wh_pro_stunde integral of  last hour (end state)
-        """
+        """hour.
 
+        akku_soc_pro_stunde begin of the hour, initial hour state!
+        last_wh_pro_stunde integral of  last hour (end state)
+        """
```
Comment (michaelosthege) [inline src/akkudoktoreos/class_ems.py:61]:
```suggestion
        """Simulate energy usage for given profiles.
        
        Args:
            start_stunde
        """
```

Comment (b0661) [reply src/akkudoktoreos/class_ems.py:61]: Hello @michaelosthege,

I only did changes to the format of the comments to pass the ruff rules for comments. Content change should be part of another PR by the author or somebody who knows the topic.

**Pull request merged**